bofire/strategies/predictives/botorch.py

```diff
@@ -332,91 +360,187 @@ def _optimize_acqf_continuous(
+        )
+
+        # check if we should take the local step
+        if self.local_search_config.is_local_step(  # type: ignore
+            local_acqf_val.item(), global_acqf_val.item()
+        ):
+            return self._postprocess_candidates(candidates=local_candidates)
+
+        # otherwise we return a set of experiments that take us to the global
+        # candidates from the last experiment ran on the equipment
+        sp = ShortestPathStrategy(
+            data_model=ShortestPathStrategyDataModel(
                 domain=self.domain,
-                n_candidates=candidate_count,
-            )
-            candidates, acqf_vals = optimize_acqf(
-                acq_function=acqfs[0],
-                bounds=bounds,
-                q=candidate_count,
-                num_restarts=self.num_restarts,
-                raw_samples=self.num_raw_samples,
-                equality_constraints=get_linear_constraints(
-                    domain=self.domain,
-                    constraint=LinearEqualityConstraint,
-                )
-                + interpoints,
-                inequality_constraints=get_linear_constraints(
-                    domain=self.domain,
-                    constraint=LinearInequalityConstraint,
-                ),
-                fixed_features=fixed_features,
-                nonlinear_inequality_constraints=nonlinear_constraints,  # type: ignore
-                return_best_only=True,
-                options=self._get_optimizer_options(),  # type: ignore
-                ic_generator=ic_generator,
-                **ic_gen_kwargs,
+                start=self.experiments.iloc[-1].to_dict(),
+                end=self._postprocess_candidates(candidates).iloc[-1].to_dict(),
             )
-        return candidates, acqf_vals
+        )
+        step = pd.DataFrame(sp.step(sp.start)).T
+        return pd.concat((step, self.predict(step)), axis=1)
 
     def _ask(self, candidate_count: int) -> pd.DataFrame:  # type: ignore
-        """[summary]
+        """Ask the strategy for new experiments.
 
         Args:
-            candidate_count (int, optional): [description]. Defaults to 1.
-
-        Returns:
-            pd.DataFrame: [description]
-
+            candidate_count (int): The batch size of new experiments
+                requested from the strategy. Defaults to 1.
         """
         assert candidate_count > 0, "candidate_count has to be larger than zero."
         if self.experiments is None:
             raise ValueError("No experiments have been provided yet.")
 
+        if (
+            self.trust_region_config is not None
+            and not self.trust_region_config.has_sufficient_experiments(
+                self.experiments, self.domain
+            )
+        ):
+            reference_experiment = self.experiments.loc[
+                self.trust_region_config.X_center_idx
+            ]
+            # instantiate a random strategy and ask it for the new experiments.
```

Comment:
I propose to move this as a method to the trust region data model, that receives a domain and spits out the local domain, which we then here path to the random strategy to sample new solutions. What do you think?

Reply:
I took a quick look it this and it will introduce circular dependencies